sanic/request.py

```diff
@@ -347,6 +351,27 @@ def token(self):
 
         return auth_header
 
+    @property
+    def credentials(self) -> Credentials:
+        """Attempt to return the auth header value.
+
+        Covers NoAuth, Basic Auth, Bearer Token, Api Token authentication schemas.
+
+        :return: A named tuple with token or username and password related to request
+        """
+        prefixes = ("Basic", "Bearer", "Token")
+        auth_header = self.headers.getone("authorization", None)
+
+        if auth_header is not None:
+            for prefix in prefixes:
+                if prefix in auth_header:
+                    header_value = auth_header.partition(prefix)[-1].strip()
+                    if prefix == "Basic":
+                        username, password = b64decode(header_value.encode("utf-8")).decode().split(":")
+                        return Credentials(username=username, password=password, token=None)
+                    return Credentials(token=header_value, username=None, password=None)
+        return Credentials(token=auth_header, username=None, password=None)
```

Comment:
I wonder if it makes sense to take this parsing out of the `Request` and into the model itself:

```python
credentials = Credentials.parse(auth_header)
```

This would allow it to be much more easily extended by application developers or users as discussed.